src/lib/es2022.array.d.ts

```diff
@@ -0,0 +1,103 @@
+interface Array<T> {
+    /**
+     * Returns the item located at the specified index.
+     * @param index The zero-based index of the desired code unit. A negative index will count back from the last item.
+     */
+    at(index?: number): T;

```

Comment:
> I can't find any discussion from TC39 that says this is intentional, though. Happy to see one if exists.

I don't think there was explicit discussion, but certainly as a member of TC39 it is my view that `at` coerces `undefined` to `0` only because everything similar in the language works that way, not because we actually think people should pass `undefined` or omit the parameter. By analogy, you can also call it with `'0'`, but that doesn't mean `| '0'` belongs in the signature.

As a user, if I ever saw `at` being called with `undefined` or without an argument, I'd assume that was a bug.